### ISSUE TITLE
About  CPU Offloading

### ISSUE THREAD

**Issue by lhj-git** [OpenBMB/BMTrain#63]
If I wanna use CPU Offloading in my model, which API should I call in BMTrain?

**Comment by a710128** [comment]
Use bmtrain.optim.AdamOffloadOptimizer.

See https://github.com/OpenBMB/BMTrain/blob/0.2.0/example/train.py#L54

**Comment by eggiter** [comment]
@a710128 Does our implementation of `AdamOffloadOptimizer` work together with ZeRO optimization by **offloading** only partitioned optimizer states and **updating** only partitioned parameters belonged to current rank?

**Comment by Achazwl** [comment]
It works together with ZeRO. Only partitioned parameters are stored on each rank, so in adam_offload.py you cannot find related codes. The parameter is already partitioned in https://github.com/OpenBMB/BMTrain/blob/0.2.0/bmtrain/parameter.py#L56, while gradient also disperses during backpropagation https://github.com/OpenBMB/BMTrain/blob/0.2.0/bmtrain/block_layer.py#L240

**Comment by eggiter** [comment]
got it.
